packages/babel-plugin-transform-destructuring/test/api.js

```diff
@@ -0,0 +1,70 @@
+import { parse } from "@babel/parser";
+import { transformFromAstSync } from "@babel/core";
+import _traverse from "@babel/traverse";
+const traverse = _traverse.default;
+
+import path from "path";
+import { fileURLToPath } from "url";
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+import glob from "glob";
+import { promises as fs } from "fs";
+
+const files = await new Promise((resolve, reject) => {
+  glob(
+    path.resolve(__dirname, "fixtures/**/@(exec|input).js"),
+    function (err, files) {
+      if (err) reject(err);
+      else resolve(files);
+    },
+  );
+});
+
+describe("regression-14438", function () {
+  describe.each(files)("fixture: %s", file => {
+    let src, originalAst, transformed;
+
+    beforeEach(async () => {
+      src = await fs.readFile(file, "utf-8");
+
+      originalAst = parse(src, {
+        allowImportExportEverywhere: true,
+        allowAwaitOutsideFunction: true,
+        allowReturnOutsideFunction: true,
+        allowSuperOutsideMethod: true,
+        allowUndeclaredExports: true,
+        errorRecovery: true,
+        attachComment: false,
+        createParenthesizedExpressions: true,
+      });
+
+      transformed = transformFromAstSync(originalAst, src, {
+        plugins: [
+          [
+            "@babel/plugin-transform-destructuring",
+            { useBuiltIns: true, loose: true },
+          ],
+        ],
+        configFile: false,
+        ast: true,
+        code: true,
+      });
+    });
```

Comment:
We can have `babel-helper-fixture` run this plugin for us.

1. create or update `./options.json` under "fixtures/allowArrayLike", "fixtures/destructuring" and other level 2 fixture folders.
2. add `"plugins": ["relative/path/to/checkScopeInfo.js"]` to `options.json`.

Now the test will run with the new plugin _and_ the designated test options (instead of providing our own).

Reply:
I see the point now though. If it's just to be added at `fixtures/destructuring` etc level then it's certainly less work, and this is certainly quite a useful check to just generalize it to more plugins.

Can you give me an example of somewhere else where tests are run like this though, so I can get a better idea of how to do it?

Reply:
> Can you give me an example of somewhere else where tests are run like this though, so I can get a better idea of how to do it? 
`packages/babel-plugin-transform-modules-commonjs/test/fixtures/interop/multi-load` is an example that a custom plugin is used in a test.